ocaml/xapi/xapi_vgpu_type.ml

```diff
@@ -467,36 +467,53 @@ module Intel = struct
       List.nth device.Pci.Pci_dev.size 2
       |> Int64.of_nativeint
     in
-    List.map
+    let rec filter_none acc lst =
+      match lst with
```

Comment:
I like the first option. My concern with the second is a mix between readability and optimality. I don't like to filter and unbox options in two steps, but it's only a matter of taste. But it's a few time now that I need a `filter_map (f : F -> T option) (lst: list F) : list T`. I am thinking about adding it to our `stdext`.

Also, I want to make few tests to see if the compiler optimizes the chain of maps or not, although it does not really make any difference for this case.